internal/service/elasticsearch/domain.go

```diff
@@ -65,6 +65,33 @@ func ResourceDomain() *schema.Resource {
 				}
 				return true
 			}),
+			customdiff.ForceNewIf("encrypt_at_rest.0.enabled", func(_ context.Context, d *schema.ResourceDiff, meta interface{}) bool {
+				newVersion := d.Get("elasticsearch_version").(string)
+
+				if newVersion < "6.7" {
```

Comment:
I suggest using `gversion.NewVersion` and `gversion.LessThan` here rather than string comparison.
e.g. https://github.com/hashicorp/terraform-provider-aws/blob/8bde0a26b57ccd1fd0c6c29dffaf9967269235a8/internal/service/elasticsearch/domain.go#L884-L890

Reply:
Nice! 👍